### ISSUE TITLE
Sample and OTU ordering do not work for heatmap but do work for barplot

### ISSUE THREAD

**Issue by shan-e-s** [microbiome/microbiome#210]
I am trying to plot a heatmap using the code:

p1<-plot_composition(ps, plot.type = "heatmap", sample.sort = sorted.names, otu.sort = otu.order)

where the sorted.names variable is a character vector of sample names and the otu.order variable is a character vector of taxa names.

Specifying both of these variables has no effect on the sample or otu order of a heatmap however they are ordered correctly when using the same code but specifying plot.type="barplot".

I tried a hacky solution of directly changing the ggplot's internal values:
p1[["plot_env"]][["order.cols"]] <- otu.order
but this did not work

EDIT:
i am also getting this warning which must be related as the values correspond to the number of samples and taxa respectively:

Warning: 'length(x) = 201 > 1' in coercion to 'logical(1)'Warning: 'length(x) = 16 > 1' in coercion to 'logical(1)'

**Comment by antagomir** [comment]
Thanks for reporting! 

Could you post a example reproducing the bug? This could be done with one of our demo data sets, e.g. `data(dietswap)`. Would ensure a timely and correct treatment of the bug.

**Comment by shan-e-s** [comment]
Sorry for the delay.

When i try to plot a heatmap with the demo data set i get the error: 

Error in hclust(as.dist(1 - cor(mat, use = "pairwise.complete.obs"))) :
NA/NaN/Inf in foreign function call (arg 10)

**Comment by shan-e-s** [comment]
Sorry for the delay.

When i try to plot a heatmap with the demo data set i get the error:

Error in hclust(as.dist(1 - cor(mat, use = "pairwise.complete.obs"))) :
NA/NaN/Inf in foreign function call (arg 10)

data(dietswap)
microbiome::plot_composition(dietswap, plot.type = "heatmap")

**Comment by antagomir** [comment]
Ok - 

some of the taxonomic features have zero std:
`sum(apply(otu_table(dietswap), 1, sd)==0)`

This leads errors when trying to calculate distances.

You could focus on features with >0 prevalence, filtering out nondetected / rare taxa, e.g.
`dietswap <- core(dietswap, detection = 1, prevalence = 0)`

This will then give you a working heatmap:

```
library(microbiome)
data(dietswap)
dietswap <- core(dietswap, detection = 1, prevalence = 0)
microbiome::plot_composition(dietswap, plot.type = "heatmap")
```

We should probably add an informative error message of something.

**Comment by shan-e-s** [comment]
Ah perfect, thanks so much for getting back to me. Apologies for not getting around to the reprex- im in the final month of my phd so things are a bit manic here.

All the best,
Shane
